### PR TITLE
feat: Add campaigns in web widget

### PR DESCRIPTION
**Pull request by muhsin-k**
- Fetch campaigns 
- Store campaigns
- Timer for executing campaigns
- API call to events endpoint

## How to Test
1. Enable inbox campaigns

Open `app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue`
Add the code on line number 337

```
{
 key: 'campaign',
 name: this.$t('INBOX_MGMT.TABS.CAMPAIGN'),
}
```
2. Go to any website inbox

3. Click on the settings

4. Go to the campaigns tab

5. Create a new campaign by entering details title, messages, sender, URL and time on page

6. Open the webpage which configured website inbox

7. Wait for the time specified in the campaign

8. Check messages are coming over the widget

### CODE REVIEW

File: app/javascript/widget/App.vue [hunk 1]
```diff
@@ -86,7 +87,7 @@ export default {
     },
     setLocale(locale) {
       const { enabledLanguages } = window.chatwootWebChannel;
-      if (enabledLanguages.some(lang => lang.iso_639_1_code === locale)) {
+      if (enabledLanguages.some((lang) => lang.iso_639_1_code === locale)) {
```
Comment (nithindavid) [inline app/javascript/widget/App.vue:90]: There are some syntax changes with parenthesis of arrow functions, which is not aligned to other parts of code. Can you revert them ?

File: app/javascript/widget/api/campaign.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+};
+
+const triggerCampaign = async ({ campaignId }) => {
+  const { websiteToken } = window.chatwootWebChannel;
+  const urlData = endPoints.triggerCampaign(websiteToken, campaignId);
+  await API.post(
+    urlData.url,
+    { ...urlData.data },
+    {
+      params: {
```
Comment (nithindavid) [inline app/javascript/widget/api/campaign.js:17]: We can move this inside the endPoints function.

File: app/javascript/widget/helpers/campaignTimer.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+      trigger_rules: { time_on_page: timeOnPage },
+      id: campaignId,
+    } = campaign;
+    setTimeout(async () => {
+      await triggerCampaign({ campaignId });
+    }, timeOnPage * 1000);
+  });
+};
+
+export default { start };
```
Comment (nithindavid) [inline app/javascript/widget/helpers/campaignTimer.js:14]: Can we omit the default export for helpers ? so `import { start } from 'widget/helpers/campaignTimer';` would work.

**Review by nithindavid — CHANGES_REQUESTED** [review]
Please see the comments.

**Comment by pranavrajs** [comment]
@muhsin-k Would you please share the test cases and assign the PR to someone for testing it?

**Review by nithindavid — DISMISSED** [review]

**Review by sojan-official — APPROVED** [review]

**Pull request merged**